BSP/Tests/ADC.c

```diff
@@ -0,0 +1,390 @@
+  AdcHandle.Init.EOCSelection          = ADC_EOC_SINGLE_CONV;           /* EOC flag picked-up to indicate conversion end */
+  AdcHandle.Init.LowPowerAutoWait      = DISABLE;                       /* Auto-delayed conversion feature disabled */
+  AdcHandle.Init.ContinuousConvMode    = ENABLE;                        /* Continuous mode enabled (automatic conversion restart after each conversion) */
+  AdcHandle.Init.NbrOfConversion       = 1;                             /* Parameter discarded because sequencer is disabled */
+  AdcHandle.Init.DiscontinuousConvMode = DISABLE;                       /* Parameter discarded because sequencer is disabled */
+  AdcHandle.Init.NbrOfDiscConversion   = 1;                             /* Parameter discarded because sequencer is disabled */
+  AdcHandle.Init.ExternalTrigConv      = ADC_SOFTWARE_START;            /* Software start to trig the 1st conversion manually, without external event */
+  AdcHandle.Init.ExternalTrigConvEdge  = ADC_EXTERNALTRIGCONVEDGE_NONE; /* Parameter discarded because software trigger chosen */
+  AdcHandle.Init.DMAContinuousRequests = ENABLE;                        /* ADC DMA continuous request to match with DMA circular mode */
+  AdcHandle.Init.Overrun               = ADC_OVR_DATA_OVERWRITTEN;      /* DR register is overwritten with the last conversion result in case of overrun */
+  AdcHandle.Init.OversamplingMode      = DISABLE;                       /* No oversampling */
+  /* Initialize ADC peripheral according to the passed parameters */
+  if (HAL_ADC_Init(&AdcHandle) != HAL_OK)
+  {
+    Error_Handler();
+  }
+  
+  
+  /* ### - 2 - Start calibration ############################################ */
+  if (HAL_ADCEx_Calibration_Start(&AdcHandle, ADC_SINGLE_ENDED) !=  HAL_OK)
+  {
+    Error_Handler();
+  }
+  
+  /* ### - 3 - Channel configuration ######################################## */
+  sConfig.Channel      = ADCx_CHANNEL;                /* Sampled channel number */   
+      // ADCx_CHANNEL is another thing defined in main.h 
+  sConfig.Rank         = ADC_REGULAR_RANK_1;          /* Rank of sampled channel number ADCx_CHANNEL */
+  sConfig.SamplingTime = ADC_SAMPLETIME_6CYCLES_5;   /* Sampling time (number of clock cycles unit) */
+  sConfig.SingleDiff   = ADC_SINGLE_ENDED;            /* Single-ended input channel */
+  sConfig.OffsetNumber = ADC_OFFSET_NONE;             /* No offset subtraction */ 
+  sConfig.Offset = 0;                                 /* Parameter discarded because offset correction is disabled */
+  if (HAL_ADC_ConfigChannel(&AdcHandle, &sConfig) != HAL_OK)
+  {
+    Error_Handler();
+  }
+  
+  /* ### - 4 - Start conversion in DMA mode ################################# */
+  if (HAL_ADC_Start_DMA(&AdcHandle,
+                        (uint32_t *)aADCxConvertedData,
+                        ADC_CONVERTED_DATA_BUFFER_SIZE
+                       ) != HAL_OK)
+  {
+    Error_Handler();
+  }
+  
+  /* Infinite Loop */
+  while (1)
+  { 
+  }
+}
```

Comment:
Where do you read and print out the ADC data? It looks like you're mostly just configuring the pins to be ADC.